h/static/scripts/directive/test/annotation-share-dialog-test.js

```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+var angular = require('angular');
+
+var util = require('./util');
+
+describe('annotationShareDialog', function () {
+  var element;
+
+  function getCopyBtn() {
+    return element.find('.annotation-share-dialog-link__btn');
+  }
+
+  before(function () {
+    angular.module('app', [])
+      .directive('annotationShareDialog',
+        require('../annotation-share-dialog'));
+  });
+
+  beforeEach(function () {
+    angular.mock.module('app');
+  });
+
+  describe('The annotation share dialog', function () {
+    it('has class is-open set when it is open', function () {
+      element = util.createDirective(document, 'annotationShareDialog', {
+        isOpen: true
+      });
+
+      assert.isOk(element.find('.annotation-share-dialog').hasClass('is-open'));
+    });
+
+    it('does not have class is-open set when it is not open', function () {
+      element = util.createDirective(document, 'annotationShareDialog', {
+        isOpen: false
+      });
+
+      assert.isNotOk(element.find('.annotation-share-dialog').hasClass('is-open'));
+    });
+  });
+
+  describe('vm.copyToClipboard()', function () {
+    var stub;
+
+    beforeEach(function () {
+      stub = sinon.stub(document, 'execCommand');
+      element = util.createDirective(document,
+        'annotationShareDialog',
+        {
+          group: {
+            name: 'Public',
+            type: 'public',
+            public: true
+          },
+          uri: 'fakeURI',
+          isPrivate: false,
+        }
+      );
+    });
+
+    afterEach(function () {
+      stub.restore();
+    });
+
+    it('sets success class on successful copy to clipboard', function () {
+      var expectedMessage = 'Link copied to clipboard!';
+
+      getCopyBtn().click();
+
+      var actualMessage = element.find('.annotation-share-dialog-link__feedback').text();
+      assert.isAbove(actualMessage.indexOf(expectedMessage), -1);
+    });
+
+    it('unsets success class after one second on successful copy to clipboard', function () {
+      var clock = sinon.useFakeTimers();
+      var expectedMessage = 'Link copied to clipboard!';
+
+      getCopyBtn().click();
+
+      clock.tick(1999);
+      clock.restore();
+
+      var actualMessage = element.find('.annotation-share-dialog-link__feedback').text();
+      assert.isBelow(actualMessage.indexOf(expectedMessage), 0);
```

Comment:
Won't block merge on this but in future I think it would be clearer to write tests for the presence/absence of a substring by comparing for equality/inequality to -1 rather than above/below zero.